src/DataWrangling/ecco_metadata.jl

```diff
@@ -0,0 +1,173 @@
+using CFTime
+using Dates
+import Dates: year, month, day
+import Oceananigans.Fields: set!
+import Base
+
+struct ECCO2Monthly end
+struct ECCO2Daily end
+struct ECCO4Monthly end
+
+# Metadata holding the ECCO dataset information:
+# - `name`: The name of the dataset.
+# - `dates`: The dates of the dataset, in a `AbstractCFDateTime` format.
+# - `version`: The version of the dataset, could be ECCO2Monthly, ECCO2Daily, or ECCO4Monthly.
```

Comment:
why are all the versions "ECCO2" but the type here is ECCO?

Reply:
there is one ECCO4Montly version also.